invenio_rdm_records/resources/serializers/datacite/schema.py

```diff
@@ -222,6 +222,13 @@ def _read_description_type(self, id_):
             system_identity
         )
 
+    def _read_date_type(self, id_):
```

Comment:
shall we take the oportunity and refactor this function as a generic one (i.e. a function receiving a `vocabulary_type` plus `id_` and doing the query? so we can remove the rest of the functions?

Feel free to shelve it for another issue 👼

Reply:
Indeed, this was partially done here <https://github.com/inveniosoftware/invenio-rdm-records/pull/704>, we are still missing affiliations. Will refactor this one (`date_type`) in this PR, thanks!